paint/data/dataset.py

```diff
@@ -0,0 +1,437 @@
+import json
+import logging
+from concurrent.futures import ThreadPoolExecutor, as_completed
+from pathlib import Path
+from typing import Any, Union
+
+import cv2
+import pandas as pd
+import torch
+from torch.utils.data import Dataset
+from torchvision import transforms
+from tqdm import tqdm
+from typing_extensions import Self
```

Comment:
Should this be selected based on the Python version? If so, everything before 3.11 should use typing_extensions and everything afterwards should use typing

Reply:
Done :)